extensions/qcom/GLSL_QCOM_image_processing2.txt

```diff
@@ -0,0 +1,212 @@
+    |                                      | sum of the squares of the values (SSD).     |
+    +--------------------------------------+---------------------------------------------+
+    | vec4 textureBlockMatchWindowSADQCOM( | Windowed block matching operation that      |
+    |       sampler2D target               | measures the correlation (or similarity)    |
+    |       uvec2     targetCoord,         | of the target block and reference block.    |
+    |       sampler2D reference,           | targetCoord and refCoord specify the bottom-|
+    |       uvec2     refCoord,            | left corner of the block in target and      |
+    |       uvec2     blockSize)           | reference images. The error metric is the   |
+    |                                      | sum of the absolute values (SAD).           |
+    +--------------------------------------+---------------------------------------------+
+    | vec4 textureBlockMatchGatherSSDQCOM( | Gathered block matching operation that      |
+    |       sampler2D target               | measures the correlation (or similarity)    |
+    |       uvec2     targetCoord,         | of the target block and reference block.    |
+    |       sampler2D reference,           | targetCoord and refCoord specify the bottom-|
+    |       uvec2     refCoord,            | left corner of the block in target and      |
+    |       uvec2     blockSize)           | reference images. The error metric is the   |
+    |                                      | sum of the squares of the values (SSD).     |
+    +--------------------------------------+---------------------------------------------+
+    | vec4 textureBlockMatchGatherSADQCOM( | Gathered block matching operation that      |
+    |       sampler2D target               | measures the correlation (or similarity)    |
+    |       uvec2     targetCoord,         | of the target block and reference block.    |
+    |       sampler2D reference,           | targetCoord and refCoord specify the bottom-|
+    |       uvec2     refCoord,            | left corner of the block in target and      |
+    |       uvec2     blockSize)           | reference images. The error metric is the   |
+    |                                      | sum of the absolute values (SAD).           |
+    +--------------------------------------+---------------------------------------------+
+
+
+    * target and reference specify the two input textures. If
+      the block from one texture is going to be used repeatedly across multiple
+      instructions, that texture should be specified as the referenceTexture.
+
+    * targetCoord and referenceCoord specify the texture coordinates (in texels)
+      of the bottom left corner of the block.
+
+    * blockSize specifies the size (in texels) of the blocks to be compared.
+
+
+Modifications to the OpenGL ES Shading Language Specification, Version 3.10, dated 29 January 2016
```

Comment:
If these are the same as the modifications for the desktop shading language, let's not write them all out again. I think we normally have a section "Interactions with the OpenGL ES Shading Language" that notes any differences between the two. If there are no differences then that section wouldn't be needed.

Reply:
removed